engine/flink/avro-util/src/test/scala/pl/touk/nussknacker/engine/avro/source/KafkaAvroSourceFactorySpec.scala

```diff
@@ -189,13 +189,32 @@ class KafkaAvroSourceFactorySpec extends KafkaAvroSpecMixin with KafkaAvroSource
       case LatestSchemaVersion => SchemaVersionOption.LatestOptionName
       case ExistingSchemaVersion(version) => version.toString
     }
+    val state = validateParamsAndInitializeState(sourceFactory, topic, version)
     val source = sourceFactory
-      .implementation(Map(KafkaAvroBaseTransformer.TopicParamName -> topic, KafkaAvroBaseTransformer.SchemaVersionParamName -> version),
-        List(TypedNodeDependencyValue(metaData), TypedNodeDependencyValue(nodeId)), None)
+      .implementation(
+        Map(KafkaAvroBaseTransformer.TopicParamName -> topic, KafkaAvroBaseTransformer.SchemaVersionParamName -> version),
+        List(TypedNodeDependencyValue(metaData), TypedNodeDependencyValue(nodeId)),
+        state)
       .asInstanceOf[Source[AnyRef] with TestDataGenerator with FlinkSourceTestSupport[AnyRef] with ReturningType]
 
     source.returnType shouldEqual AvroSchemaTypeDefinitionExtractor.typeDefinition(expectedSchema)
 
     source
   }
+
+  // Use final contextTransformation to 1) validate parameters and 2) to calculate the final state.
+  // This transformation can return
+  // - the state that contains information on runtime key-value schemas, which is required in createSource.
+  // - validation errors
+  private def validateParamsAndInitializeState(sourceFactory: KafkaAvroSourceFactory[Any], topic: String, version: String): Option[KafkaAvroSourceFactory.KafkaAvroSourceFactoryState] = {
```

Comment:
Can you return `Validated` or at least `Try` instead of trhowing exceptions?

Reply:
Fixed, returned `Validated`.